src/shared/tpm2-util.c

```diff
@@ -1997,13 +2132,25 @@ int tpm2_extend_bytes(
 }
 #endif
 
-int tpm2_parse_pcrs(const char *s, uint32_t *ret) {
-        const char *p = ASSERT_PTR(s);
+char *tpm2_pcr_mask_to_string(uint32_t mask) {
+        _cleanup_free_ char *s = NULL;
+
+        FOREACH_PCR_IN_MASK(n, mask)
+                if (strextendf_with_separator(&s, "+", "%d", n) < 0)
+                        return NULL;
+
```

Comment:
note that this means that if mask is zero, we'll return NULL here. which makes oom conditions indistinguishable from empty masks...

hence, maybe add:

```c
if (!s)
        return strdup("");
```

here

Reply:
will do, thanks